### PR TITLE
HWKAPM-757 Enable custom byteman rules to be defined

### PR DESCRIPTION
**Pull request by objectiser**

### CODE REVIEW

**Comment by objectiser** [comment]
@jpkrohling Could you review this please?

Simply allows end user to include custom byteman (.btm) rule files in the classpath of their application to perform additional instrumentation.

**Comment by jpkrohling** [comment]
LGTM!

File: client/agent-opentracing/src/main/java/org/hawkular/apm/agent/opentracing/OpenTracingRuleLoader.java [hunk 1]
```diff
@@ -39,10 +42,16 @@
  */
 public class OpenTracingRuleLoader {
 
+    /**  */
```
Comment (pavolloffay) [inline client/agent-opentracing/src/main/java/org/hawkular/apm/agent/opentracing/OpenTracingRuleLoader.java]: Could you please remove this auto-generated comments? At line 48 too.

File: tests/agent-opentracing/src/test/java/org/hawkular/apm/tests/agent/opentracing/custom/CustomRuleITest.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+import org.hawkular.apm.tests.common.ClientTestBase;
+import org.hawkular.apm.tests.common.Wait;
+import org.junit.Test;
+
+/**
+ * @author gbrown
+ */
+public class CustomRuleITest extends ClientTestBase {
+
+    /**  */
```
Comment (pavolloffay) [inline tests/agent-opentracing/src/test/java/org/hawkular/apm/tests/agent/opentracing/custom/CustomRuleITest.java]: auto generated comment

File: client/agent-opentracing/src/main/java/org/hawkular/apm/agent/opentracing/OpenTracingRuleLoader.java [hunk 3]
```diff
@@ -89,4 +82,39 @@ public static void initialize(Retransformer trans) throws Exception {
...
+        String entryName = uri.toString();
+
+        int separator = entryName.indexOf(PATH_JAR_SEPARATOR);
+        if (separator != -1) {
+            fs = FileSystems.newFileSystem(URI.create(entryName.substring(0, separator)),
+                    Collections.<String, Object>emptyMap());
+            entryName = entryName.substring(separator + PATH_JAR_SEPARATOR.length());
+        } else if (entryName.startsWith(FILE_SCHEME)) {
+            fs = FileSystems.getFileSystem(URI.create(FILE_SCHEME + File.separator));
+            entryName = entryName.substring(5);
```
Comment (pavolloffay) [inline client/agent-opentracing/src/main/java/org/hawkular/apm/agent/opentracing/OpenTracingRuleLoader.java]: Could you please change this to `FILE_SCHEME.length()` (IIRC)

**Comment by objectiser** [comment]
retest this please

**Comment by objectiser** [comment]
@jpkrohling  @pavolloffay Thanks!

**Pull request merged**